sunpy/net/dataretriever/sources/goes.py

```diff
@@ -222,184 +84,108 @@ class SUVIClient(GenericClient):
+    3 Results from the SUVIClient:
+         Start Time           End Time      Instrument ... Level   Wavelength
+    ------------------- ------------------- ---------- ... ----- --------------
+    2020-07-10 00:00:00 2020-07-10 00:04:00       SUVI ...     2 304.0 Angstrom
+    2020-07-10 00:04:00 2020-07-10 00:08:00       SUVI ...     2 304.0 Angstrom
+    2020-07-10 00:08:00 2020-07-10 00:12:00       SUVI ...     2 304.0 Angstrom
+    <BLANKLINE>
+    <BLANKLINE>
 
-        if check_var_count == 1:
-            return True
+    """
+    baseurl1b = (r'https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes'
+                 r'{SatelliteNumber}/l1b/suvi-l1b-{elem:2}{wave:03}/%Y/%m/%d/OR_SUVI-L1b.*\.fits.gz')
+    pattern1b = ('{}/goes/goes{SatelliteNumber:2d}/l{Level:2w}/suvi-l1b-{}{Wavelength:03d}/'
+                 '{year:4d}/{month:2d}/{day:2d}/{}_s{:7d}{hour:2d}{minute:2d}{second:2d}'
+                 '{:1d}_e{:7d}{ehour:2d}{eminute:2d}{esecond:2d}{:1d}_{}')
+    baseurl2 = (r'https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes{SatelliteNumber}/'
+                r'l2/data/suvi-l2-ci{wave:03}/%Y/%m/%d/dr_suvi-l2-ci{wave:03}_g{SatelliteNumber}_s%Y%m%dT%H%M%SZ_.*\.fits')
+    pattern2 = ('{}/goes/goes{SatelliteNumber:2d}/{}/dr_suvi-l{Level}-ci{Wavelength:03d}_g{SatelliteNumber:2d}_s'
+                '{year:4d}{month:2d}{day:2d}T{hour:2d}{minute:2d}{second:2d}Z_e'
+                '{eyear:4d}{emonth:2d}{eday:2d}T{ehour:2d}{eminute:2d}{esecond:2d}Z_{}')
+    optional = {a.Wavelength, a.Source, a.Provider, a.Level, a.Physobs, a.goes.SatelliteNumber}
+
+    def pre_search_hook(cls, *args, **kwargs):
+        matchdict = cls._get_match_dict(*args, **kwargs)
+        return cls.baseurl1b, cls.pattern1b, cls.baseurl2, cls.pattern2, matchdict
+
+    def post_search_hook(self, i, matchdict):
+
+        # extracting start times and end times
+        start = datetime(i['year'], i['month'], i['day'], i['hour'], i['minute'], i['second'])
+        end = datetime(i['year'], i['month'], i['day'], i['ehour'], i['eminute'], i['esecond'])
+        timerange = TimeRange(start, end)
+
+        map_ = OrderedDict()
+        map_['Time'] = timerange
+        map_['Start Time'] = start.strftime(TIME_FORMAT)
+        map_['End Time'] = end.strftime(TIME_FORMAT)
+        map_['Instrument'] = matchdict['Instrument'][0].upper()
+        map_['Physobs'] = matchdict['Physobs'][0]
+        map_['Source'] = matchdict['Source'][0]
+        map_['Provider'] = matchdict['Provider'][0]
+        map_['SatelliteNumber'] = i['SatelliteNumber']
+        map_['Level'] = i['Level']
+        map_['Wavelength'] = i['Wavelength']*u.Angstrom
+        map_['url'] = i['url']
+        return map_
+
+    def search(self, *args, **kwargs):
+        supported_waves = [94, 131, 171, 195, 284, 304]*u.Angstrom
+        all_waves = []
+        baseurl1b, pattern1b, baseurl2, pattern2, matchdict = self.pre_search_hook(*args, **kwargs)
```

Comment:
why even bother with a call to `pre_search_hook` in this client?

Reply:
You're right, not calling now.